source/logrepl/cdc.go

```diff
@@ -124,9 +153,45 @@ func (i *CDCIterator) NextN(ctx context.Context, n int) ([]opencdc.Record, error
 		return nil, fmt.Errorf("n must be greater than 0, got %d", n)
 	}
 
-	var recs []opencdc.Record
+	// First, we check if there are any records from the previous batch
+	// that we can start with.
+	recs := make([]opencdc.Record, len(i.recordsForNextRead), n)
```

Comment:
It'd be interesting to see if this PR implementation fixes https://github.com/ConduitIO/conduit-connector-postgres/issues/282.